app/models/widget.rb

```diff
@@ -3,29 +3,34 @@ class Widget < ActiveRecord::Base
 
   validates :sm_size, :md_size, :lg_size, :kind, presence: true
   validates :mailchimp_segment_id, uniqueness: true, allow_nil: true
+
   belongs_to :block
+
   has_one :community, through: :mobilization
   has_one :mobilization, through: :block
+
   has_many :form_entries
   has_many :donations
   has_many :matches
   has_many :activist_pressures
+
   store_accessor :settings
 
-  after_create :async_create_mailchimp_segment
   delegate :user, to: :mobilization
 
   def as_json(*)
     WidgetSerializer.new(self, {root: false})
   end
 
   def segment_name
+    kinds_correlation = {'pressure' => 'P', 'form' => 'F', 'match' => 'M'}
```

Comment:
Donation is not a Mailchimpable -- it's not synchronized wih mailchimp. Will we turn it a mailchimp synchronizable entity?